ST_mortality.c

```diff
@@ -68,70 +71,74 @@ void _kill_maxage(void);
+ *  is captured by the probability of establishment (see
+ *  rgroup_Establish()).  However, for consistency with other
+ *  arrays, the age-related arrays are indexed from 0.  Thus,
+ *  age is base1 but the arrays are base0.
+ * 
+ *  The outline for this function is:
+ *    1) Death by limited resources.
+ *    2) Age independent mortality.
+ *    3) Slow growth mortality.
+ *    4) Succulent wet season mortality.
+ *    5) Death by pats, mounds or burrows.
+ *
+ *  More specifically:
+ *  - Process each group separately.
+ *  - If resources required for a group are greater than available
+ *    (PR > 1) for the maximum years stretching is allowed, kill
+ *    plants according the rules for insufficient resources.  If
+ *    PR < 1, reset number of stretched years to zero.  This means
+ *    that even one year of adequate resources nulls any number of
+ *    years of stretched resources.
+ *  - For each species, if the user requested the mortality functions
+ *    to be executed, run the age-independent and slow-growth
+ *    routines.
+ *  - If the current species is a succulent and this is a wet year,
+ *    execute the mortality function for succulents.  That is,
+ *    reduce each plant by some constant proportion, defined by
+ *    eqn 16.  The parameters are defined in the group-parms and
+ *    the reduction amount is computed in the Env module based
+ *    on precipitation.
+ *  - execute disturbance effects, if any.
+ * 
+ * \param killed Reference to a boolean. Set to TRUE if one or more
+ *               individuals were killed in any species. FALSE otherwise.
+ * 
+ * \sideeffect Individuals are killed. Individuals are stored in a
+ *             linked list in Species[sp]->IndivHead, so expect the 
+ *             list to be modified for all sp.\n
+ *             _SomeKillage will be set to TRUE if any individuals die.
+ * 
+ * \sa mort_EndOfYear()
+ * \sa rgroup_Establish()
+ * 
+ * \ingroup MORTALITY
+ */
 void mort_Main( Bool *killed) {
-/*======================================================*/
-/* PURPOSE */
-/* This routine contains all the references to mortality
-   that occurs during the growing season.  (See mort_EndOfYear()
-   for other routines.)  It reduces "plant-space" according to
-   resource availability, age, growth rate, and disturbance
-   conditions.  Also, succulents are "mortified" if it's a wet
-   year.
-
-   An important consideration is the fact that the age of new
-   plants starts at 1.  Consider that for it to establish, it
-   has already survived the massive mortality of year 0; this
-   rate of year 0 mortality, among other factors of survival,
-   is captured by the probability of establishment (see
-   rgroup_Establish()).  However, for consistency with other
-   arrays, the age-related arrays are indexed from 0.  Thus,
-   age is base1 but the arrays are base0.
-
-   ALGORITHM
-   The outline of the steps is:
-     _no_resources(group)  - eqns 7, 8, 9
-     _age_independent(spp) - eqn 14
```

Comment:
I would like to retain the reference to eqns in the above comments and add a reference to Coffin and Lauenroth 1990 when doing so.

Reply:
Addressed by commit 984bd13.